mesonbuild/msetup.py

```diff
@@ -172,10 +174,8 @@ def validate_dirs(self, dir1: str, dir2: str, reconfigure: bool, wipe: bool) ->
                       'using the same options as passed when configuring the build.'
                       '\nTo change option values, run "meson configure" instead.')
                 raise SystemExit
-        else:
-            has_cmd_line_file = os.path.exists(coredata.get_cmd_line_file(build_dir))
-            if (wipe and not has_cmd_line_file) or (not wipe and reconfigure):
-                raise SystemExit(f'Directory does not contain a valid build tree:\n{build_dir}')
+        elif not has_partial_build:
```

Comment:
... this should update `self.options.wipe = self.options.reconfigure = False`.

Reply:
I don't think that's a good idea, a partial build (a previous failed setup) could contain all kind of generated files that `--wipe` should still remove.

Reply:
@eli-schwartz does that answer your review  comment? ^

Reply:
Yes, sorry, that's correct.